storage/client/src/samples/java/blob/FileTransferExample.java

```diff
@@ -0,0 +1,171 @@
+        SharedKeyCredential credential = new SharedKeyCredential(accountName, accountKey);
+
+        /**
+         * From the Azure portal, get your Storage account blob service URL endpoint.
+         * The URL typically looks like this:
+         */
+        String endPoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
+
+        // Create a StorageClient object that wraps the service endpoint, credential and a request pipeline.
+        // Now you can use the storageClient to perform various container and blob operations.
+        StorageClient storageClient = StorageClient.storageClientBuilder().endpoint(endPoint).credential(credential).buildClient();
+
+
+        // This example shows several common operations just to get you started.
+
+        /*
+        Create a client that references a to-be-created container in your Azure Storage account. This returns a
+        ContainerClient object that wraps the container's endpoint, credential and a request pipeline (inherited from storageClient).
+        Note that container names require lowercase.
+         */
+        ContainerClient containerClient = storageClient.getContainerClient("myjavacontainerparallelupload" + System.currentTimeMillis());
+
+        // Create a container in Storage blob account.
+        containerClient.create();
+        /*
+        Create a client that references a to-be-created blob in your Azure Storage account's container.
+        This returns a BlockBlobClient object that wraps the blob's endpoint, credential and a request pipeline
+        (inherited from containerClient). Note that blob names can be mixed case.
+         */
+        // Create a BlockBlobClient object that wraps a blob's endpoint and a default pipeline.
+        String filename = "BigFile.bin";
+        BlockBlobClient blobClient = containerClient.getBlockBlobClient(filename);
+
+        // Create the empty uploadFile and downloadFile.
+        URL folderUrl = FileTransferExample.class.getClassLoader().getResource(".");
+        File dirPath = new File(folderUrl.getPath() + LARGE_TEST_FOLDER);
+        File largeFile = null;
+        File downloadFile = null;
+        if (dirPath.exists() || dirPath.mkdir()) {
+            largeFile = new File(folderUrl.getPath() + LARGE_TEST_FOLDER + filename);
+            downloadFile = new File(folderUrl.getPath() + LARGE_TEST_FOLDER + "downloadFile.bin");
+        } else {
+            System.err.println("Failed to create the large file dir.");
+        }
+
+        // Generate random things to uploadFile, which makes the file with size of 100MB.
+        Long fileSize = 100 * 1024 * 1024L;
+        try {
+            RandomAccessFile raf = new RandomAccessFile(largeFile, "rw");
+            raf.setLength(fileSize);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to generate a large file! Error details: " + e.getMessage());
+        }
+
+        // Upload the large file to storage blob.
```

Comment:
Maybe a note that this method is optimized to upload the file as fast as possible in parallel.

Reply:
Done.